tables/process.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+import pandas as pd
+
+from tables.table import Table
+
+
+class TableProcessor:
+    """
+    Processes tables based on specified rules.
+
+    This class provides methods to process tables extracted by the TableExtractor.
+    It allows customization of header handling, column removal, and other transformations.
+
+    Attributes:
+        num_header_rows (int, optional): Number of header rows (default is 1).
+        ignored_non_header_rows (list[int], optional): List of row indices to ignore as non-header rows (default is
+           None).
+        merged_header_rows (list[int], optional): List of row indices where headers are merged (default is None).
+        col_names_to_drop (list[str], optional): List of column names to drop (default is None).
+        drop_empty_rows (bool, optional): Whether to drop empty rows (default is False).
+        drop_empty_tables (bool, optional): Whether to drop tables that are entirely empty (default is False).
+        dropdown_placeholder (str, optional): Placeholder text for dropdowns (default is "< Select >").
+
+    Methods:
+        process(self, table: Table) -> None:
+            Processes the given table based on specified rules.
+
+    Example usage:
+    >>> processor = TableProcessor(num_header_rows=2, col_names_to_drop=["Unused", "Extra"])
+    >>> table = Table(...)  # Assume you have a Table instance
+    >>> processor.process(table)
+    """
+
+    num_header_rows: int
+    ignored_non_header_rows: list[int]
+    merged_header_rows: list[int]
+    drop_empty_rows: bool
+    drop_empty_tables: bool
+
+    def __init__(
+        self,
+        num_header_rows: int = 1,
+        ignored_non_header_rows: list[int] | None = None,
+        merged_header_rows: list[int] | None = None,
+        col_names_to_drop=None,
+        drop_empty_rows: bool = False,
+        drop_empty_tables: bool = False,
+        dropdown_placeholder: str = "< Select >",
+    ):
+        if ignored_non_header_rows is None:
```

Comment:
Just double-checking - is the reason we can't just have an empty list default argument the idea that the list object is created once and re-used across instances, so is susceptible to weirdness?